src/dbg/threading.h

```diff
@@ -96,9 +110,18 @@ class SectionLockerGlobal
         if(m_SRWLocks)
         {
             if(Shared)
+            {
                 m_ReleaseSRWLockShared(&m_srwLocks[LockIndex]);
-            else
+                return;
+            }
+
+            assert(m_owner[LockIndex].count && m_owner[LockIndex].thread);
+            m_owner[LockIndex].count--;
+            if(m_owner[LockIndex].count == 0)

```

Comment:
Doesn't this require locking because no atomic operations are used to manipulate or compare these count values? This means you have a race condition on this count variable.